dotnet/src/Azure.Iot.Operations.Protocol/State/PropertyConsumer.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Iot.Operations.Protocol.Models;
+using Azure.Iot.Operations.Protocol.RPC;
+using Azure.Iot.Operations.Protocol.Telemetry;
+
+namespace Azure.Iot.Operations.Protocol.State
+{
+    public abstract class PropertyConsumer<TProp, TBool> : IAsyncDisposable
+        where TProp : class
+        where TBool : class
+    {
+        private bool _isDisposed;
+
+        public PropertyWriteRequester<TProp, TBool> PropertyWriteRequester { get; }
+
+        public PropertyReadRequester<TProp, TBool> PropertyReadRequester { get; }
+
+        public PropertyWatchRequester<TBool> PropertyWatchRequester { get; }
```

Comment:
Nit: You refer to this pattern as observe/unobserve in the ADR, so these fields should probably be

```csharp
        public PropertyObserveRequester<TBool> PropertyObserveRequester { get; }
```

Reply:
Yes, indeed.  I updated the [ADR PR](https://github.com/Azure/iot-operations-sdks/pull/922) this morning with the new terms, but I've not yet updated this PR or the [Rust SDK PR](https://github.com/Azure/iot-operations-sdks/pull/944).